src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1897,6 +1988,100 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
             }
         }
 
+        private bool AdoptOrphanedFor(DocumentsOperationContext context, string id)
+        {
+            using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            {
+                var collectionName = GetCollectionFor(context, lowerIdPrefix);
+                if (collectionName == null)
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                    return false;
+                }
+
+
+                if (ShouldCreateDeleteRevision(context, lowerId, lowerIdPrefix, collectionName, out var table, out var lastRevision))
+                {
+                    var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                    var newEtag = _database.DocumentsStorage.GenerateNextEtag();
+                    var cv = _documentsStorage.GetNewChangeVector(context, newEtag);
+                    CreateDeleteRevision(context, table, id, collectionName, cv, lastModifiedTicks, lastRevision.Flags);
```

Comment:
`CreateDeletedRevision`

Reply:
why not to go through the `Delete` method?

Reply:
because the 'Delete' method is doing other things besides creating a deletion revision..
I wanted only to create the 'Delete Revision' without actually deleting the revisions using the right configuration, without checking conflicts, etc..

Reply:
It actually called 'Delete Revision' in the studio so I went with that name

Reply:
It is actually called 'Delete Revision' in the studio so I went with that name.
I think it is called 'delete revision' and not 'deleted revision' because this revision exists and isn't 'deleted', and mentions the deletion of the document.